src/controllers/hid/hidreportdescriptor.h

```diff
@@ -163,7 +163,7 @@ class Report {
         return m_lastBitPosition;
     }
 
-    const std::vector<Control>& getControls() const {
+    std::vector<Control>& getControls() {
         return m_controls;
     }
 
```

Comment:
This commit is overzealous. I'm sorry my previous comment was misleading. `const &` are fine (and even preferred to non-const& in order to uphold class invariants) only the `const std::vector` from `getListOfReports` was unnecessary since in that case, you're building a whole new vector locally and then transferring it to the caller (in which case the class doesn't need to care about that vector anymore). Does that make sense?

Reply:
If you need mutable access members too, either add non-const overloads or use C++23 "deducing `this`" feature.

Reply:
Hopefully addressed correct now

Reply:
I haven't double checked whether this applies to other member functions too. Assuming it doesn't then yes, this is addressed.